src/components/EnhancedTable/EnhancedTable.jsx

```diff
@@ -49,150 +60,414 @@ class EnhancedTable extends React.Component {
-      x => (Immutable.List.isList(orderBy) ? x.getIn(orderBy) : x.get(orderBy)),
-      sortComparator(numericSort, orderDirection)
-    );
-  }
+    if (_.isObject(rowProps)) {
+      return rowProps;
+    }
 
-  validateColumnKeys() {
-    const { columns } = this.props;
+    return {};
+  };
 
-    const missingKeys = columns
-      .filter(x => !x.has('key'))
-      .map(x => x.get('title'))
-      .toSet();
+  validateColumnKeys(columns) {
+    const missingKeys = _.map(_.filter(columns, x => !_.has(x, 'key')), x => x.label);
 
-    if (missingKeys.size > 0) {
+    if (missingKeys.length > 0) {
       throw `The key property is required and must be unique for each column configured. Affected columns ${missingKeys.join(
         ','
       )}`;
     }
   }
 
-  render() {
-    const { orderBy, orderDirection, page, rowsPerPage } = this.state;
-
-    const {
-      columns,
-      dataSource,
-      rowKey,
-      rowProps,
-      className,
-      loading,
-      classes,
-      rowSelection,
-      fullHeight,
-      noDataText,
-      rowRenderOptions
-    } = this.props;
-
-    this.validateColumnKeys();
+  getSelectedRowCount = () => {
+    const { rowSelection, rowKey } = this.props;
+    const { data } = this.state;
+
+    const rowKeyFn = _.isFunction(rowKey) ? rowKey : defaultRowKey;
+
+    if (rowSelection && _.isFunction(rowSelection.get('rowIsSelected'))) {
+      return _.filter(data, x => rowSelection.get('rowIsSelected')(x, rowKeyFn(x))).length;
+    }
+
+    return 0;
+  };
+
+  noRowsRenderer = () => {
+    const { noDataText, rowHeight, classes } = this.props;
 
     return (
-      <div className={classNames(classes.tableContainer, { [classes.tableContainerFullHeight]: fullHeight })}>
-        {loading && <Loading />}
-        <div className={classes.tableScrollArea}>
-          <Table className={className}>
-            <EnhancedTableHead
-              columns={columns}
-              onRequestSort={this.handleRequestSort}
-              orderBy={orderBy}
-              orderDirection={orderDirection}
-              dataSource={dataSource}
-              rowSelection={rowSelection}
-              rowKey={rowKey}
-            />
-            <EnhancedTableBody
-              columns={columns}
-              dataSource={this.sort(dataSource)}
-              rowKey={rowKey}
-              page={page}
-              rowsPerPage={rowsPerPage}
-              rowSelection={rowSelection}
-              rowProps={rowProps}
-              noDataText={noDataText}
-              rowRenderOptions={rowRenderOptions}
+      <TableCell
+        style={{ height: `${rowHeight}px` }}
+        component="div"
+        variant="body"
+        className={classNames(classes.tableCellFlex, classes.centerFlex)}
+      >
+        {noDataText}
+      </TableCell>
+    );
+  };
+
+  getRowKeys = data => {
+    const { rowKey } = this.props;
+    return _.isFunction(rowKey) ? _.map(data, rowKey) : [];
+  };
+
+  cellRenderer = cellRenderData => {
```

Comment:
We may want to break this out at some point, at >100 lines it is pretty beefy sitting inside of here